### DISCUSSION TITLE
remove sp dependency in ufs

### DISCUSSION THREAD

**Discussion by junwang-noaa** [ufs-community/ufs-weather-model#1081] [Announcements]
## Description
Currently the ccpp-physics has dependency on splat from sp library. The dependency can be removed. 

## Solution
Two solutions are proposed:
1) remove the Gaussian grid support for the fixed files read in the ccpp-physics. Currently model has the capability to read in the fixed files on the native tiled grid. So if model switches to use the tiled files, the dependency of sp lib can be removed.
2) the splat used in ccpp-physics from sp lib is a small Fortran subroutine without further dependency on any other file and it is available in the fv3/io. It is possible to use one version of splat shared by both ccpp-physics and fv3/io. This way the Gaussian grid fixed file will still be supported in ccpp-physics.

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. 
- Update update will be in ccpp-physics.

**Comment by climbfuji** [comment]
Solution (2) replaces the sp lib dependency with another dependency, unless you want to move the routine into ccpp-physics and then use it in fv3atm (I don't think we should do this, because it is really a shared dependency).

Another option would be to duplicate the splat routine and add it to `sfcsub.F`, but I do not like this approach either.

I prefer solution (1) if we are sure that we don't need to support reading Gaussian grid fixed files in ccpp-physics in the future.

**Comment by DusanJovic-NOAA** [comment]
These latlon fixed files are used by the model (according to control regression test):
```
FNAISC   latlon   CFSR.SEAICE.1982.2012.monthly.clim.grb
FNALBC2  latlon   global_albedo4.1x1.grb
FNGLAC   latlon   global_glacier.2x2.grb
FNMXIC   latlon   global_maxice.2x2.grb
FNVMXC   latlon   global_shdmax.0.144x0.144.grb
FNVMNC   latlon   global_shdmin.0.144x0.144.grb
FNSLPC   latlon   global_slope.1x1.grb
FNSNOC   latlon   global_snoclim.1.875.grb
FNTG3C   latlon   global_tg3clim.2.6x1.5.grb
FNVEGC   latlon   global_vegfrac.0.144.decpercent.grb
         latlon   global_zorclim.1x1.grb
FNTSFC   latlon   RTGSST.1982.2012.monthly.clim.grb
```
and these Gaussian fixed files:
```
FNABSC   gaussian global_mxsnoalb.uariz.t126.384.190.rg.grb
FNMSKH   gaussian global_slmask.t1534.3072.1536.grb
FNALBC   gaussian global_snowfree_albedo.bosu.t126.384.190.rg.grb
FNSMCC   gaussian global_soilmgldas.statsgo.t1534.3072.1536.grb
         gaussian global_soilmgldas.t126.384.190.grb
FNSOTC   gaussian global_soiltype.statsgo.t126.384.190.rg.grb
FNVETC   gaussian global_vegtype.igbp.t126.384.190.rg.grb
```
Maybe we can just convert these gaussian to latlon. Do they have to be on Gaussian grid? I doubt.

**Comment by junwang-noaa** [comment]
Those Gaussian fixed files have already been updated and in netcdf format.

**Comment by SMoorthi-emc** [comment]
Well tile files can't be used in non-fv3 models, such as SCM!

On Mon, Jul 12, 2021 at 2:32 PM Jun Wang ***@***.***> wrote:

> Those Gaussian fixed files have already been updated and in netcdf format.
>
> On Mon, Jul 12, 2021 at 2:22 PM Dusan Jovic ***@***.***>
> wrote:
>
> > These latlon fixed files are used by the model (according to control
> > regression test):
> >
> > FNAISC latlon CFSR.SEAICE.1982.2012.monthly.clim.grb
> > FNALBC2 latlon global_albedo4.1x1.grb
> > FNGLAC latlon global_glacier.2x2.grb
> > FNMXIC latlon global_maxice.2x2.grb
> > FNVMXC latlon global_shdmax.0.144x0.144.grb
> > FNVMNC latlon global_shdmin.0.144x0.144.grb
> > FNSLPC latlon global_slope.1x1.grb
> > FNSNOC latlon global_snoclim.1.875.grb
> > FNTG3C latlon global_tg3clim.2.6x1.5.grb
> > FNVEGC latlon global_vegfrac.0.144.decpercent.grb
> > latlon global_zorclim.1x1.grb
> > FNTSFC latlon RTGSST.1982.2012.monthly.clim.grb
> >
> > and these Gaussian fixed files:
> >
> > FNABSC gaussian global_mxsnoalb.uariz.t126.384.190.rg.grb
> > FNMSKH gaussian global_slmask.t1534.3072.1536.grb
> > FNALBC gaussian global_snowfree_albedo.bosu.t126.384.190.rg.grb
> > FNSMCC gaussian global_soilmgldas.statsgo.t1534.3072.1536.grb
> > gaussian global_soilmgldas.t126.384.190.grb
> > FNSOTC gaussian global_soiltype.statsgo.t126.384.190.rg.grb
> > FNVETC gaussian global_vegtype.igbp.t126.384.190.rg.grb
> >
> > Maybe we can just convert these gaussian to latlon. Do they have to be on
> > Gaussian grid? I doubt.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878494674
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TL75Y3CGANEHPQMEHTTXMXFNANCNFSM5AHC562A
> >
> > .
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878500580>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYV63SPXOJT3FFYAD4LTXMYJ5ANCNFSM5AHC562A>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

**Comment by SMoorthi-emc** [comment]
sfcsub.F is designed to run for any random sets of grid-points as long as
latitude/longitudes are defined.
That is how it works for SCM, basically a set of random grid columns.
This functionality requires to use either lat/lon or Gaussian grid grib1
files as input; netcdf files will not work.


On Mon, Jul 12, 2021 at 2:34 PM Shrinivas Moorthi - NOAA Federal <
***@***.***> wrote:

> Well tile files can't be used in non-fv3 models, such as SCM!
>
> On Mon, Jul 12, 2021 at 2:32 PM Jun Wang ***@***.***> wrote:
>
>> Those Gaussian fixed files have already been updated and in netcdf format.
>>
>> On Mon, Jul 12, 2021 at 2:22 PM Dusan Jovic ***@***.***>
>> wrote:
>>
>> > These latlon fixed files are used by the model (according to control
>> > regression test):
>> >
>> > FNAISC latlon CFSR.SEAICE.1982.2012.monthly.clim.grb
>> > FNALBC2 latlon global_albedo4.1x1.grb
>> > FNGLAC latlon global_glacier.2x2.grb
>> > FNMXIC latlon global_maxice.2x2.grb
>> > FNVMXC latlon global_shdmax.0.144x0.144.grb
>> > FNVMNC latlon global_shdmin.0.144x0.144.grb
>> > FNSLPC latlon global_slope.1x1.grb
>> > FNSNOC latlon global_snoclim.1.875.grb
>> > FNTG3C latlon global_tg3clim.2.6x1.5.grb
>> > FNVEGC latlon global_vegfrac.0.144.decpercent.grb
>> > latlon global_zorclim.1x1.grb
>> > FNTSFC latlon RTGSST.1982.2012.monthly.clim.grb
>> >
>> > and these Gaussian fixed files:
>> >
>> > FNABSC gaussian global_mxsnoalb.uariz.t126.384.190.rg.grb
>> > FNMSKH gaussian global_slmask.t1534.3072.1536.grb
>> > FNALBC gaussian global_snowfree_albedo.bosu.t126.384.190.rg.grb
>> > FNSMCC gaussian global_soilmgldas.statsgo.t1534.3072.1536.grb
>> > gaussian global_soilmgldas.t126.384.190.grb
>> > FNSOTC gaussian global_soiltype.statsgo.t126.384.190.rg.grb
>> > FNVETC gaussian global_vegtype.igbp.t126.384.190.rg.grb
>> >
>> > Maybe we can just convert these gaussian to latlon. Do they have to be
>> on
>> > Gaussian grid? I doubt.
>> >
>> > —
>> > You are receiving this because you authored the thread.
>> > Reply to this email directly, view it on GitHub
>> > <
>> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878494674
>> >,
>> > or unsubscribe
>> > <
>> https://github.com/notifications/unsubscribe-auth/AI7D6TL75Y3CGANEHPQMEHTTXMXFNANCNFSM5AHC562A
>> >
>> > .
>> >
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878500580>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ALLVRYV63SPXOJT3FFYAD4LTXMYJ5ANCNFSM5AHC562A>
>> .
>>
>
>
> --
> Dr. Shrinivas Moorthi
> Research Meteorologist
> Modeling and Data Assimilation Branch
> Environmental Modeling Center / National Centers for Environmental
> Prediction
> 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> Tel: (301)683-3718
>
> e-mail: ***@***.***
> Phone:  (301) 683-3718   Fax: (301) 683-3718
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

**Comment by junwang-noaa** [comment]
@Moorthi-emc  Do you mean the ccpp-physics SCM or something else?

**Comment by SMoorthi-emc** [comment]
I don't know much about ccpp-SCM.  However, it used to work for all earlier
versions of NCEP/SCM (including the version Fnglin worked with many years
ago).
I don't see why it should not work with ccpp-SCM.
Actually, it can work with any grid-point models.

**Comment by junwang-noaa** [comment]
The subsfc.F in current ccpp-physics can work with the tile grid netcdf
file, I think it should work with single column model unless you are using
a different version of subsfc.F.

**Comment by SMoorthi-emc** [comment]
Jun,
   Yo are incorrect.  "netcdf" files are read as a full tile, not to an
individual grid point.
Moorthi

On Mon, Jul 12, 2021 at 2:54 PM Jun Wang ***@***.***> wrote:

> The subsfc.F in current ccpp-physics can work with the tile grid netcdf
> file, I think it should work with single column model unless you are using
> a different version of subsfc.F.
>
> On Mon, Jul 12, 2021 at 2:48 PM SMoorthi-emc ***@***.***>
> wrote:
>
> > I don't know much about ccpp-SCM. However, it used to work for all
> earlier
> > versions of NCEP/SCM (including the version Fnglin worked with many years
> > ago).
> > I don't see why it should not work with ccpp-SCM.
> > Actually, it can work with any grid-point models.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878510613
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TNE7A27O7K5BYATTRTTXM2GJANCNFSM5AHC562A
> >
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878514436>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYXUPZCIZB2DM7QEL4DTXM26ZANCNFSM5AHC562A>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

**Comment by climbfuji** [comment]
Note that gcycle/sfcsub are not used with CCPP-SCM right now

**Comment by SMoorthi-emc** [comment]
Also,  SCM grid points need not be on a "fv3" grid.  It could be any
location on the globe.
One more point to consider is to run the model using analysis as input;
which is important in making AMIP style runs.
Arun Kumar's group at CPC is already trying to do this with a version of
FV3.  Disabling this capability is not progress.

On Mon, Jul 12, 2021 at 3:00 PM Dom Heinzeller ***@***.***>
wrote:

> Note that gcycle/sfcsub are not used with CCPP-SCM right now
>
> > On Jul 12, 2021, at 12:57 PM, SMoorthi-emc ***@***.***> wrote:
> >
> >
> > Jun,
> > Yo are incorrect. "netcdf" files are read as a full tile, not to an
> > individual grid point.
> > Moorthi
> >
> > On Mon, Jul 12, 2021 at 2:54 PM Jun Wang ***@***.***> wrote:
> >
> > > The subsfc.F in current ccpp-physics can work with the tile grid netcdf
> > > file, I think it should work with single column model unless you are
> using
> > > a different version of subsfc.F.
> > >
> > > On Mon, Jul 12, 2021 at 2:48 PM SMoorthi-emc ***@***.***>
> > > wrote:
> > >
> > > > I don't know much about ccpp-SCM. However, it used to work for all
> > > earlier
> > > > versions of NCEP/SCM (including the version Fnglin worked with many
> years
> > > > ago).
> > > > I don't see why it should not work with ccpp-SCM.
> > > > Actually, it can work with any grid-point models.
> > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878510613
> > > >,
> > > > or unsubscribe
> > > > <
> > >
> https://github.com/notifications/unsubscribe-auth/AI7D6TNE7A27O7K5BYATTRTTXM2GJANCNFSM5AHC562A
> > > >
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878514436
> >,
> > > or unsubscribe
> > > <
> https://github.com/notifications/unsubscribe-auth/ALLVRYXUPZCIZB2DM7QEL4DTXM26ZANCNFSM5AHC562A
> >
> > > .
> > >
> >
> >
> > --
> > Dr. Shrinivas Moorthi
> > Research Meteorologist
> > Modeling and Data Assimilation Branch
> > Environmental Modeling Center / National Centers for Environmental
> > Prediction
> > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > Tel: (301)683-3718
> >
> > e-mail: ***@***.***
> > Phone: (301) 683-3718 Fax: (301) 683-3718
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub <
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878516123>,
> or unsubscribe <
> https://github.com/notifications/unsubscribe-auth/AB5C2RK5GYVWLLIKUJXWLYDTXM3JHANCNFSM5AHC562A
> >.
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878517831>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYR2OQWVMHLKQETQ4EDTXM3TZANCNFSM5AHC562A>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

**Comment by junwang-noaa** [comment]
Dom, how does the CCPP-SCM get fixed data at each grid point?

**Comment by climbfuji** [comment]
@grantfirl will be able to answer this question far better than I

**Comment by junwang-noaa** [comment]
Moorthi, what analysis file is this? Does the analysis have to be on the
Gaussian grid?

**Comment by yangfanglin** [comment]
I agree with Moorthi. The model should support AMIP-type
experiments which use observed SSTs, usually saved either on Gaussian grid
or la-tlon grid,  as input.  To a minimum, the support for input on global
lat-lon grids needs to be maintained.

Fanglin

On Mon, Jul 12, 2021 at 3:04 PM SMoorthi-emc ***@***.***>
wrote:

> Also, SCM grid points need not be on a "fv3" grid. It could be any
> location on the globe.
> One more point to consider is to run the model using analysis as input;
> which is important in making AMIP style runs.
> Arun Kumar's group at CPC is already trying to do this with a version of
> FV3. Disabling this capability is not progress.
>
> On Mon, Jul 12, 2021 at 3:00 PM Dom Heinzeller ***@***.***>
> wrote:
>
> > Note that gcycle/sfcsub are not used with CCPP-SCM right now
> >
> > > On Jul 12, 2021, at 12:57 PM, SMoorthi-emc ***@***.***> wrote:
> > >
> > >
> > > Jun,
> > > Yo are incorrect. "netcdf" files are read as a full tile, not to an
> > > individual grid point.
> > > Moorthi
> > >
> > > On Mon, Jul 12, 2021 at 2:54 PM Jun Wang ***@***.***> wrote:
> > >
> > > > The subsfc.F in current ccpp-physics can work with the tile grid
> netcdf
> > > > file, I think it should work with single column model unless you are
> > using
> > > > a different version of subsfc.F.
> > > >
> > > > On Mon, Jul 12, 2021 at 2:48 PM SMoorthi-emc ***@***.***>
> > > > wrote:
> > > >
> > > > > I don't know much about ccpp-SCM. However, it used to work for all
> > > > earlier
> > > > > versions of NCEP/SCM (including the version Fnglin worked with many
> > years
> > > > > ago).
> > > > > I don't see why it should not work with ccpp-SCM.
> > > > > Actually, it can work with any grid-point models.
> > > > >
> > > > > —
> > > > > You are receiving this because you authored the thread.
> > > > > Reply to this email directly, view it on GitHub
> > > > > <
> > > >
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878510613
> > > > >,
> > > > > or unsubscribe
> > > > > <
> > > >
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TNE7A27O7K5BYATTRTTXM2GJANCNFSM5AHC562A
> > > > >
> > > > > .
> > > > >
> > > >
> > > > —
> > > > You are receiving this because you commented.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878514436
> > >,
> > > > or unsubscribe
> > > > <
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYXUPZCIZB2DM7QEL4DTXM26ZANCNFSM5AHC562A
> > >
> > > > .
> > > >
> > >
> > >
> > > --
> > > Dr. Shrinivas Moorthi
> > > Research Meteorologist
> > > Modeling and Data Assimilation Branch
> > > Environmental Modeling Center / National Centers for Environmental
> > > Prediction
> > > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > > Tel: (301)683-3718
> > >
> > > e-mail: ***@***.***
> > > Phone: (301) 683-3718 Fax: (301) 683-3718
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub <
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878516123
> >,
> > or unsubscribe <
> >
> https://github.com/notifications/unsubscribe-auth/AB5C2RK5GYVWLLIKUJXWLYDTXM3JHANCNFSM5AHC562A
> > >.
> > >
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878517831
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALLVRYR2OQWVMHLKQETQ4EDTXM3TZANCNFSM5AHC562A
> >
> > .
> >
>
>
> --
> Dr. Shrinivas Moorthi
> Research Meteorologist
> Modeling and Data Assimilation Branch
> Environmental Modeling Center / National Centers for Environmental
> Prediction
> 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> Tel: (301)683-3718
>
> e-mail: ***@***.***
> Phone: (301) 683-3718 Fax: (301) 683-3718
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878520778>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY5N2L4EEVUBFS73MUDEGTTXM4DXANCNFSM5AHC562A>
> .
>


-- 
*Fanglin Yang, Ph.D.*
*Chief, Model Physics Group*
*Modeling and Data Assimilation Branch*

*NOAA/NWS/NCEP Environmental Modeling Center*

*https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/
<https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/>*

**Comment by SMoorthi-emc** [comment]
When you make AMIP runs, you normally provide observed SST analysis,
sea-ice analysis and snow analysis, generally daily analysis.
These need to be provided as grib1 files on either lat/lon or Gassian grid
- grib1 files are self-describing.
sfcsub.F can use this analysis and interpolate to any grid points provided
via the argument - so it can be used in any models, including SCM.
It does not have the capability to ingest netcdf analysis files, unless
someone is willing to write appropriate code.


On Mon, Jul 12, 2021 at 3:08 PM Jun Wang ***@***.***> wrote:

> Moorthi, what analysis file is this? Does the analysis have to be on the
> Gaussian grid?
>
> On Mon, Jul 12, 2021 at 3:04 PM SMoorthi-emc ***@***.***>
> wrote:
>
> > Also, SCM grid points need not be on a "fv3" grid. It could be any
> > location on the globe.
> > One more point to consider is to run the model using analysis as input;
> > which is important in making AMIP style runs.
> > Arun Kumar's group at CPC is already trying to do this with a version of
> > FV3. Disabling this capability is not progress.
> >
> > On Mon, Jul 12, 2021 at 3:00 PM Dom Heinzeller ***@***.***>
> > wrote:
> >
> > > Note that gcycle/sfcsub are not used with CCPP-SCM right now
> > >
> > > > On Jul 12, 2021, at 12:57 PM, SMoorthi-emc ***@***.***> wrote:
> > > >
> > > >
> > > > Jun,
> > > > Yo are incorrect. "netcdf" files are read as a full tile, not to an
> > > > individual grid point.
> > > > Moorthi
> > > >
> > > > On Mon, Jul 12, 2021 at 2:54 PM Jun Wang ***@***.***> wrote:
> > > >
> > > > > The subsfc.F in current ccpp-physics can work with the tile grid
> > netcdf
> > > > > file, I think it should work with single column model unless you
> are
> > > using
> > > > > a different version of subsfc.F.
> > > > >
> > > > > On Mon, Jul 12, 2021 at 2:48 PM SMoorthi-emc ***@***.***>
> > > > > wrote:
> > > > >
> > > > > > I don't know much about ccpp-SCM. However, it used to work for
> all
> > > > > earlier
> > > > > > versions of NCEP/SCM (including the version Fnglin worked with
> many
> > > years
> > > > > > ago).
> > > > > > I don't see why it should not work with ccpp-SCM.
> > > > > > Actually, it can work with any grid-point models.
> > > > > >
> > > > > > —
> > > > > > You are receiving this because you authored the thread.
> > > > > > Reply to this email directly, view it on GitHub
> > > > > > <
> > > > >
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878510613
> > > > > >,
> > > > > > or unsubscribe
> > > > > > <
> > > > >
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AI7D6TNE7A27O7K5BYATTRTTXM2GJANCNFSM5AHC562A
> > > > > >
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > You are receiving this because you commented.
> > > > > Reply to this email directly, view it on GitHub
> > > > > <
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878514436
> > > >,
> > > > > or unsubscribe
> > > > > <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYXUPZCIZB2DM7QEL4DTXM26ZANCNFSM5AHC562A
> > > >
> > > > > .
> > > > >
> > > >
> > > >
> > > > --
> > > > Dr. Shrinivas Moorthi
> > > > Research Meteorologist
> > > > Modeling and Data Assimilation Branch
> > > > Environmental Modeling Center / National Centers for Environmental
> > > > Prediction
> > > > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > > > Tel: (301)683-3718
> > > >
> > > > e-mail: ***@***.***
> > > > Phone: (301) 683-3718 Fax: (301) 683-3718
> > > > —
> > > > You are receiving this because you commented.
> > > > Reply to this email directly, view it on GitHub <
> > >
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878516123
> > >,
> > > or unsubscribe <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AB5C2RK5GYVWLLIKUJXWLYDTXM3JHANCNFSM5AHC562A
> > > >.
> > > >
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878517831
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ALLVRYR2OQWVMHLKQETQ4EDTXM3TZANCNFSM5AHC562A
> > >
> > > .
> > >
> >
> >
> > --
> > Dr. Shrinivas Moorthi
> > Research Meteorologist
> > Modeling and Data Assimilation Branch
> > Environmental Modeling Center / National Centers for Environmental
> > Prediction
> > 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> > Tel: (301)683-3718
> >
> > e-mail: ***@***.***
> > Phone: (301) 683-3718 Fax: (301) 683-3718
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878520778
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TOLJ2MD75WP3PIA6LTTXM4DVANCNFSM5AHC562A
> >
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/issues/687#issuecomment-878523830>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYXQGDQFAPTROXQOGFDTXM4TBANCNFSM5AHC562A>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

**Comment by climbfuji** [comment]
Another option that came up has to do with the "gray zone" where this functionality belongs. It is debatable whether this code (nudging towards sfc climatology) should be part of the physics or the host models. If it is agreed that the latter is the case, then the surface subcycling should be moved out of ccpp-physics into fv3atm.

**Comment by arunchawla-NOAA** [comment]
@junwang-noaa @climbfuji @yangfanglin @SMoorthi-emc I am changing this issue to a discussion as no solution has been agreed to here. Once a path forward is decided we can create an issue
